File: entities/clinic.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type ConsultationMode string

const (
	ConsultationModeInPerson ConsultationMode = "IN_PERSON"
	ConsultationModeOffline  ConsultationMode = "IN_PERSON"
	ConsultationModeVideo    ConsultationMode = "VIDEO"
	ConsultationModeBoth     ConsultationMode = "BOTH"
)

type DoctorClinic struct {
	ClinicID         uuid.UUID        `gorm:"primaryKey" json:"clinic_id"`
	DoctorID         uuid.UUID        `gorm:"not null;index" json:"doctor_id"`
	ClinicName       string           `gorm:"size:255;not null" json:"clinic_name"`
	Address          string           `gorm:"type:text;not null" json:"address"`
	City             string           `gorm:"size:100;not null;index" json:"city"`
	State            string           `gorm:"size:100;not null" json:"state"`
	Pincode          string           `gorm:"size:20;not null" json:"pincode"`
	Latitude         *float64         `json:"latitude,omitempty"`
	Longitude        *float64         `json:"longitude,omitempty"`
	ConsultationMode ConsultationMode `gorm:"size:50;default:'IN_PERSON';not null" json:"consultation_mode"`
	ConsultationFee  float64          `gorm:"type:decimal(10,2);not null" json:"consultation_fee"`
	CreatedAt        time.Time        `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt        time.Time        `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt        gorm.DeletedAt   `gorm:"index" json:"-"`
}

func (c *DoctorClinic) BeforeCreate(tx *gorm.DB) (err error) {
	if c.ClinicID == uuid.Nil {
		c.ClinicID = uuid.New()
	}
	return nil
}

File: python_backend/entities/clinic.py
import enum
import uuid
from sqlalchemy import Column, String, DateTime, Float, Enum, ForeignKey, Text, Numeric
from sqlalchemy.sql import func
from database import Base

class ConsultationMode(str, enum.Enum):
    IN_PERSON = "IN_PERSON"
    OFFLINE = "IN_PERSON"
    VIDEO = "VIDEO"
    BOTH = "BOTH"

class DoctorClinic(Base):
    __tablename__ = "doctor_clinics"

    clinic_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    clinic_name = Column(String(255), nullable=False)
    address = Column(Text, nullable=False)
    city = Column(String(100), nullable=False, index=True)
    state = Column(String(100), nullable=False)
    pincode = Column(String(20), nullable=False)
    latitude = Column(Float, nullable=True)
    longitude = Column(Float, nullable=True)
    consultation_mode = Column(Enum(ConsultationMode, name="consultation_mode_enum", native_enum=False), default=ConsultationMode.IN_PERSON, nullable=False)
    consultation_fee = Column(Numeric(10, 2), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), index=True, nullable=True)
